hubspot-style/src/test/java/com/hubspot/immutables/utils/WireSafeEnumTest.java

```diff
@@ -344,4 +346,31 @@ public void itDeserializesFromUnknownStringWithCollidingJsonAndCreator() throws
     assertThat(wrapper.asString()).isEqualTo("ABC");
     assertThat(wrapper.asEnum()).isEqualTo(Optional.empty());
   }
+
+  @Test
+  public void itThrowsForValueFromUnknownString() {
+    WireSafeEnum<RetentionPolicy> wrapper =
+        WireSafeEnum.fromJson(RetentionPolicy.class, "INVALID");
+    assertThat(wrapper.enumType()).isEqualTo(RetentionPolicy.class);
+    assertThat(wrapper.asString()).isEqualTo("INVALID");
+
+    assertThatThrownBy(wrapper::asEnumOrThrow)
+        .isInstanceOf(IllegalStateException.class)
+        .hasMessage("Value 'INVALID' is not valid for enum of type 'RetentionPolicy'. Valid values are: [CLASS, RUNTIME, SOURCE]");
+  }
+
+  @Test
+  public void itDoesNotThrowForValueFromKnownString() {
+    WireSafeEnum<RetentionPolicy> wrapper =
+        WireSafeEnum.fromJson(RetentionPolicy.class, "SOURCE");
+    assertThat(wrapper.enumType()).isEqualTo(RetentionPolicy.class);
+    assertThat(wrapper.asString()).isEqualTo("SOURCE");
+
+    try {
+      assertThat(wrapper.asEnumOrThrow())
+          .isEqualTo(RetentionPolicy.SOURCE);
+    } catch (IllegalStateException e) {
+      fail("IllegalStateException should not have been thrown", e);
+    }
```

Comment:
I think you can ditch the try/catch, the test should fail automatically if it throws an exception